ui/pages/Popup.tsx

```diff
@@ -54,6 +61,39 @@ export default function Popup({ store }: { store: Store }): ReactElement {
   const [shouldDisplayDecoy, setShouldDisplayDecoy] = useState(false)
   const [isDirectionRight, setIsDirectionRight] = useState(true)
   const [showTabBar, setShowTabBar] = useState(true)
+  const renderCount = useRef(0)
+
+  function saveHistoryEntries(routeHistoryEntities: Location[]) {
+    const isNotOnKeyringRelatedPage =
+      routeHistoryEntities[routeHistoryEntities.length - 1].pathname !==
+        "/signTransaction" &&
+      !routeHistoryEntities[routeHistoryEntities.length - 1].pathname.includes(
+        "/keyring/"
+      )
+
+    // Initial extension load takes two renders because of setting
+    // animation control states. `initialEntries` needs to be a reversed
+    // version of route history entities. Without avoiding the initial load,
+    // entries will keep reversing.
+    if (renderCount.current > 1 && isNotOnKeyringRelatedPage) {
+      const entries = routeHistoryEntities
+        .reduce((agg: Partial<Location>[], entity) => {
+          const { ...entityCopy } = entity as Partial<Location>
+          delete entityCopy.hash
+          delete entityCopy.key
+          agg.push(entityCopy)
```

Comment:
Not blocking: we can avoid the `delete` operator by destructuring. E.g.

```suggestion
          const { key: {}, hash: {}, ...entry } = entity as Partial<Location>
          agg.push(entry)
```